From c5042be29e4c4fbdcdcc18fcefc73e0dad0588d0 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:03:45 -0400
Subject: [usb] cdc-wdm: ignore speed change notifications

Message-id: <1410887463-20674-154-git-send-email-dzickus@redhat.com>
Patchwork-id: 95499
O-Subject: [RHEL7 PATCH 153/591] usb: cdc-wdm: ignore speed change notifications
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 9983d6dc4e8a8a1ce599cb426c0ec9384219b476
Author: Bjorn Mork <bjorn@mork.no>
Date:   Tue Oct 29 09:52:57 2013 +0100

    usb: cdc-wdm: ignore speed change notifications

    The only notification supported by the Device Management class is
    Response Available. But this driver is also used as a subdriver of
    other CDC classes, allowing notifications like Speed Change and
    Network Connection. This results in log messages which are only
    confusing to an end user:

     [66255.801874] cdc_mbim 1-3:1.5: unknown notification 42 received: index 5 len 8

    These drivers use cdc-wdm as a subdriver to allow access to an
    embedded management protocol, and all management is expected to
    use this protocol. There is therefore no need to handle any of
    these optional CDC notifications. Instead we can let the cdc-wdm
    driver recognize them and log a debug level message instead of an
    error.

    Reported-by: Rob Gardner <robmatic@gmail.com>
    Signed-off-by: Bjorn Mork <bjorn@mork.no>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/class/cdc-wdm.c b/drivers/usb/class/cdc-wdm.c
index 589ea58..4d38759 100644
--- a/drivers/usb/class/cdc-wdm.c
+++ b/drivers/usb/class/cdc-wdm.c
@@ -254,6 +254,10 @@ static void wdm_int_callback(struct urb *urb)
    "NOTIFY_NETWORK_CONNECTION %s network",
    dr->wValue ? "connected to" : "disconnected from");
   goto exit;
+ case USB_CDC_NOTIFY_SPEED_CHANGE:
+  dev_dbg(&desc->intf->dev, "SPEED_CHANGE received (len %u)",
+   urb->actual_length);
+  goto exit;
  default:
   clear_bit(WDM_POLL_RUNNING, &desc->flags);
   dev_err(&desc->intf->dev,
-- 
1.7.1